### DISCUSSION TITLE
Can't set PGID to existing Group Id. (Code Change suggested).

### DISCUSSION THREAD

**Discussion by Alan111S** [Taxel/PlexTraktSync#1258] [Support]
The recent changes for [Switch user only if needed (configured via PUID/PGID)](https://github.com/Taxel/PlexTraktSync/commit/d2439daa7b7f754d7e0634c6f688067a3cf875a1) have caused an issue for me running on Synology NAS.

This recently added code in entrypoint.sh tries to switch the id of the plextraktsync group to PGID set in the docker compose file:-
`if [ -n "$gid" ] && [ "$(id -g $APP_GROUP)" != "$gid" ]; then 
              groupmod -g "$gid" "$APP_GROUP"
        fi`

On Synology NAS, it's convenient to set PGID=100 which is the id of the "users" group, but the issue is the docker file already contains a group with id=100 so I get an error:-

`groupmod: GID '100' already exists`

If you could change the command to perform `groupmod -o -g "$gid" "$APP_GROUP"` then we could use a non-unique group id.

I've had a look at some other docker images I use and having duplicate group ids doesn't seem to cause an issue.

**Comment by theChad6** [comment]
I have this same problem on Synology. Should this discussion be converted to an issue or enhancement?

**Reply by Alan111S** [reply]
> I have this same problem on Synology. Should this discussion be converted to an issue or enhancement?

It's certainly easy to do that. I wanted to see how many others are affected first. In my docker-compose file I'm hard coding PGID to 1000 (which is not a real group on my Synology) but since it's not many files I can live with that but at least it's starting up now.

If taxel doesn't comment or give this any attention soon, I'll convert it to an issue early next week.

**Reply by glensc** [reply]
The created issue cross-link (for posterity):
- https://github.com/Taxel/PlexTraktSync/issues/1291

also, Taxel is inactive for years, see his zero profile activity:
- https://github.com/Taxel
